### PR TITLE
Add extension for CosmosDb/TableService that accepts TokenCredential for authentication

### PR DESCRIPTION
**Pull request by levimatheri**
**What this PR does / why we need it**:
Adds a new builder extension for registering CosmosDb/TableService health check, that accepts a `TokenCredential` for authentication using managed identities.

**Which issue(s) this PR fixes**:
Currently we have to pass a connection string to the cosmos account, which includes the authorization key. This is not best practice in Azure, and managed identity use is recommended.

Please reference the issue this PR will close: #1298, #1134

### CODE REVIEW

**Comment by sungam3r** [comment]
@levimatheri I'll merge this after fixing API approval file in tests. You need run test once, then delete ~.approved.txt file and rename ~.received.txt into ~.approved.txt.

**Comment by levimatheri** [comment]
Fixed @sungam3r

File: test/HealthChecks.CosmosDb.Tests/DependencyInjection/RegistrationTests.cs
Comment (sungam3r) [inline test/HealthChecks.CosmosDb.Tests/DependencyInjection/RegistrationTests.cs:61]:
```suggestion
                .AddCosmosDb("cosmosdbaccounturi", new MockTokenCredential(), "databasename");
```

File: test/HealthChecks.CosmosDb.Tests/DependencyInjection/RegistrationTests.cs
Comment (sungam3r) [inline test/HealthChecks.CosmosDb.Tests/DependencyInjection/RegistrationTests.cs:93]:
```suggestion
                .AddCosmosDbCollection("cosmosdbaccounturi", new MockTokenCredential(), "databasename", collections: new[] { "first-collection", "second_collections" });
```

**Review by sungam3r — APPROVED** [review]
Thanks.

**Comment by emlyn-arkahna** [comment]
Any chance we can get this released to nuget?

**Comment by sungam3r** [comment]
https://www.nuget.org/packages/AspNetCore.HealthChecks.CosmosDb/6.0.3

**Pull request merged**
